test/built-ins/Proxy/isExtensible/return-is-different-from-target.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 9.5.3
+description: >
+    Throws a TypeError exception if boolean trap result is not the same as
+    target.[[IsExtensible]]() result
+info: >
+    [[IsExtensible]] ( )
+
+    ...
+    1. Let handler be the value of the [[ProxyHandler]] internal slot of O.
+    2. If handler is null, throw a TypeError exception.
+    ...
+    4. Let target be the value of the [[ProxyTarget]] internal slot of O.
+    5. Let trap be GetMethod(handler, "isExtensible").
+    ...
+    8. Let booleanTrapResult be ToBoolean(Call(trap, handler, «target»)).
+    9. ReturnIfAbrupt(booleanTrapResult).
+    10. Let targetResult be target.[[IsExtensible]]().
+    11. ReturnIfAbrupt(targetResult).
+    12. If SameValue(booleanTrapResult, targetResult) is false, throw a TypeError exception.
+    ...
+
+---*/
+
+var p = new Proxy({}, {
+    isExtensible: function(t) {
+        return !Object.isExtensible(t);
```

Comment:
It would be more direct/readable to return the literal `false` here. It's probably also worth adding an analogous test for non-extensible objects.